specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/stable/2017-04-01/azureactivedirectory.json

```diff
@@ -0,0 +1,557 @@
+        "tags": ["DiagnosticSettingsCategories"],
+        "operationId": "DiagnosticSettingsCategory_List",
+        "description": "Lists the diagnostic settings categories for AadIam.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to view diagnostic setting categories has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get list of diagnostic setting categories",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsCategoryResourceCollection"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting categories": {
+            "$ref": "./examples/listDiagnosticSettingsCategories.json"
+          }
+        } 
+      }
+    }
+  },
+  "definitions": {
+    "ProxyOnlyResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource location"
+        },
+        "kind": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource kind"
+        },
+        "tags": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource tags"
+        },
+        "identity": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource identity"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "A proxy only azure resource object."
+    },
+    "RetentionPolicy": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "A value indicating whether the retention policy is enabled."
+        },
+        "days": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 0,
+          "description": "The number of days for the retention in days. A value of 0 will retain the events indefinitely."
+        }
+      },
+      "required": [ "enabled", "days" ],
+      "description": "Specifies the retention policy for the log."
+    },
+    "MetricSettings": {
+      "properties": {
+        "timeGrain": {
+          "type": "string",
+          "format": "duration",
+          "description": "The timegrain of the metric in ISO8601 format."
+        },
+        "category": {
```

Comment:
does the category options keep changing frequently? if not, it could be made as an enum so its easier for developers to consume, instead of first calling another API.

Reply:
I was informed we actually don't support metric settings currently, so I removed the whole section.